operator/internal/manifests/ruler_test.go

```diff
@@ -126,19 +125,41 @@ func TestNewRulerStatefulSet_MountsRulesInPerTenantIDSubDirectories(t *testing.T
 				"tenant-b": {RuleFiles: []string{"rule-a-alerts.yaml", "rule-b-recs.yaml"}},
 			},
 		},
-	})
+		RulesConfigMapNames: []string{"config"},
+	}
+	sts := manifests.NewRulerStatefulSet(opts)
 
 	vs := sts.Spec.Template.Spec.Volumes
 
-	var (
-		volumeNames []string
-		volumeItems []corev1.KeyToPath
-	)
+	var volumeNames []string
 	for _, v := range vs {
 		volumeNames = append(volumeNames, v.Name)
-		volumeItems = append(volumeItems, v.ConfigMap.Items...)
 	}
 
 	require.NotEmpty(t, volumeNames)
-	require.NotEmpty(t, volumeItems)
+}
+
+func TestNewRulerStatefulSet_ShardedRulesConfigMap(t *testing.T) {
+	// Create a large config map which will be split into 2 shards
+	opts := testOptions_withSharding()
+	rulesCMShards, err := manifests.RulesConfigMapShards(opts)
+	require.NoError(t, err)
+	require.NotNil(t, rulesCMShards)
+	require.Len(t, rulesCMShards, 2)
+
+	for _, shard := range rulesCMShards {
+		opts.RulesConfigMapNames = append(opts.RulesConfigMapNames, shard.Name)
+	}
+
+	// Create the Ruler StatefulSet and mount the ConfigMap shards into the Ruler pod
+	sts := manifests.NewRulerStatefulSet(*opts)
+
+	vs := sts.Spec.Template.Spec.Volumes
+
+	var volumeNames []string
+	for _, v := range vs {
+		volumeNames = append(volumeNames, v.Name)
+	}
+
+	require.Len(t, volumeNames, 2)
```

Comment:
Ok this checks that we still keep having 2 volumes, regardless how many configmap come in. However this is missing to check that we have one projection per configmap in the VolumeMounts.